### PR TITLE
Various digital ocean tweaks.

### PR DESCRIPTION
**Pull request by ejj**
Most of these are aimed at scaling to 64 nodes for spark.

### CODE REVIEW

**Comment by kklin** [comment]
I'll review this since I probably need it for scaling the integration tests

File: cloud/google/google.go [hunk 1]
```diff
@@ -461,6 +461,15 @@ func (prvdr *Provider) UpdateFloatingIPs(machines []db.Machine) error {
 	return nil
 }
 
+// Cleanup removes unnecessary detritus from this provider.  It's intended to be called
+// when there are no VMS running or expected to be running soon.
```
Comment (kklin) [inline cloud/google/google.go]: VMS -> VMs

File: cloud/vagrant/vagrant.go [hunk 2]
```diff
@@ -132,3 +132,9 @@ func (prvdr Provider) SetACLs(acls []acl.ACL) error {
 func (prvdr *Provider) UpdateFloatingIPs([]db.Machine) error {
 	return errors.New("vagrant provider does not support floating IPs")
 }
+
+// Cleanup removes unnecessary detritus from this provider.  It's intended to be called
+// when there are no VMS running or expected to be running soon.
```
Comment (kklin) [inline cloud/vagrant/vagrant.go]: VMS -> VMs

File: cloud/amazon/amazon.go [hunk 3]
```diff
@@ -625,6 +625,29 @@ func logACLs(add bool, perms []*ec2.IpPermission) {
 	}
```
Comment (kklin) [inline cloud/amazon/amazon.go]: In the commit message, "cloud provides" -> "cloud providers"

File: cloud/cloud.go [hunk 4]
```diff
@@ -28,6 +28,8 @@ type provider interface {
 	SetACLs([]acl.ACL) error
 
 	UpdateFloatingIPs([]db.Machine) error
+
+	Cleanup() error
```
Comment (kklin) [inline cloud/cloud.go]: I had to think a bit about the role of this function, and I think it'd be helpful if its constraints were documented more. At first I thought it was like a `Close` function where after it's been called, it won't be called again (but it'll actually be called every two minutes). It would also be helpful to document somewhere that `List` is called every two minutes, so if the provider implementation creates resources in that function for some reason, `Cleanup` won't behave properly.

Basically, I think this function imposes a very specific structure on the way cloud providers should be implemented, and that it'd be helpful for that to be documented somewhere.

Another example is that the Google provider only creates the Network in the `New` function, so if `Cleanup` removed the `Network` (which might be reasonable if it behaved like `Close`), then future `Boot` calls will fail since the network is gone.

File: cloud/google/google.go [hunk 5]
```diff
@@ -461,6 +461,15 @@ func (prvdr *Provider) UpdateFloatingIPs(machines []db.Machine) error {
...
+func (prvdr *Provider) Cleanup() error {
+	// XXX: The only thing that could need cleaning up is firewalls, but those are
```
Comment (kklin) [inline cloud/google/google.go]: FWIW the network needs to be cleaned up as well, although I'm fine with not doing that in this commit.

Comment (kklin) [reply cloud/google/google.go]: It'll take some work since I think a single network is used for all regions, rather than a network per region.

Comment (ejj) [reply cloud/google/google.go]: Yeah google is a mess ....

Comment (ejj) [reply cloud/google/google.go]: Changed the comment to this:

```+       // XXX: There are several resources that could be cleaned up here.  Some are
+       // handled in a sort of ad hoc way in the other provider functions.  Some are not
+       // cleaned up at all.```
